### DISCUSSION TITLE
Settings (override) per profile?

### DISCUSSION THREAD

**Discussion by architectingcode** [microsoft/terminal#12725] [Q&A] [answered]
Are you considering supporting a settings.json per profile to augment or override the global settings?
Example scenario: Having a custom action in certain shells only.
Let's say I add a sendInput action to clear the screen with Ctrl+L but I want that action to be available in Command Prompt and Power Shell only, so that the default behavior in Ubuntu is not modified (Ctrl+L in Linux can be used in the middle of a command and clears the screen preserving what has been typed so far). Yes, I can work around this by using another keyboard binding and only use it in the intended shells but this is just a practical use case where it would be useful.

Please excuse me if this is in the roadmap or features request already, I couldn't find it easily. 

This tool is AWESOME, I love it.

Thank you

**Comment by zadjii-msft** [comment, marked as answer]
I think you're specifically looking for #5790, which we're using to track "profile-specific keybindings". It's not really on the roadmap at the moment. We kinda need to figure out global actions (#6899), and also keymaps (#2205). Once those two are done, we need to figure out what layering keymaps looks like. Like, presumably there's a default keymap that ships with the terminal, but you probably don't want to start with _nothing_ when authoring a new keymap. Like, you probably want to just start from the default keymap and make some modifications. 

So the spec for that is a bit complicated. Since it's got a bit of prerequisite work we haven't really gotten around to committing it to a timeframe yet. 

For more discussion I'd follow up in #5790.
